app/ts/components/subcomponents/PreactBlocky.tsx

```diff
@@ -110,51 +108,51 @@ function generateIdenticon(options: BlockieProps, canvasRef: HTMLCanvasElement)
 		}
 	}
 
-	const opts = options || {}
-	const size = opts.size || 8
-	const scale = opts.scale || 4
-	const seed = opts.seed || Math.floor((Math.random()*Math.pow(10,16))).toString(16)
+	const seed = addressString(address)
 
 	seedrand(seed)
 
-	const color = opts.color || createColor()
-	const bgcolor = opts.bgColor || createColor()
-	const spotcolor = opts.spotColor || createColor()
-	const imageData = createImageData(size)
+	const color = createColor()
+	const bgcolor = createColor()
+	const spotcolor = createColor()
+	const imageData = createImageData(8)
 	const canvas = setCanvas(canvasRef, imageData, color, scale, bgcolor, spotcolor)
 
 	return canvas
 }
 
-export default function Blockie(props: BlockieProps) {
-	const scale = props.scale || 4
-	const dimension = (props.size || 8) * scale
-	const [seed, setSeed] = useState<string | undefined>(props.seed)
-	const [dataURL, setDataURL] = useState<string | undefined>(dataURLCache.get(`${ props.seed }!${ dimension }`))
-
-	useEffect(() => {
-		if (dataURL === undefined || seed !== props.seed) {
-			setSeed(props.seed)
-			const element = document.createElement('canvas')
-			generateIdenticon(props, element)
-			element.toBlob((blob) => {
-				if (!blob) return
-				const dataUrl = URL.createObjectURL(blob)
-				setDataURL(dataUrl)
-				dataURLCache.set(dataUrl, `${ props.seed }!${ dimension }`)
-			})
-		}
-	}, [props.seed])
+async function renderBlockieToUrl(address: Signal<bigint>, scale: Signal<number> | undefined) {
+	const key = `${ address.value }!${ scale?.value || 4 }`
+	const cacheResult = dataURLCache.get(key)
+	if (cacheResult !== undefined) return cacheResult
+	const future = new Future<string>()
+	const element = document.createElement('canvas')
+	generateIdenticon(address.value, scale?.value || 4, element)
+	element.toBlob((blob) => {
+		if (!blob) return
+		const dataUrl = URL.createObjectURL(blob)
+		dataURLCache.set(dataUrl, key)
+		future.resolve(dataUrl)
+	})
+	return await future
+}
+
+export function Blockie(props: BlockieProps) {
+	const dimension = 8 * (props.scale?.value || 4)
+    const { value: dataURL, waitFor } = useAsyncState<string>()
+    useSignalEffect(() => {
+		props.address.value
```

Comment:
This shouldn't be necessary, I suggested adding it just for testing but IIRC it didn't fix anything.  As long as this function touches `props.address.value` before it goes async (which it appears to do) then the signal effect will work.